aries_cloudagent/multitenant/manager_provider.py

```diff
@@ -0,0 +1,49 @@
+"""Profile manager for multitenancy."""
+
+import logging
+
+from ..config.provider import BaseProvider
+from ..config.settings import BaseSettings
+from ..config.injector import BaseInjector
+from ..config.base import InjectionError
+from .manager import MultitenantManager
+from .askar_profile_manager import AskarProfileMultitenantManager
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MultitenantManagerProvider(BaseProvider):
+    """
+    Multitenant manager provider.
+
+    Decides which manager to use based on the settings.
+    """
+
+    MANAGER_TYPES = {
+        "basic": MultitenantManager,
+        "askar-profile": AskarProfileMultitenantManager,
```

Comment:
I think this needs to use DeferLoad from utils.classloader in order to avoid importing the Askar classes until needed - as that is not currently a required dependency.

Reply:
Agree changed, thanks